src/lib/Context.ts

```diff
@@ -119,14 +147,19 @@ export function consolidateConfigs<T>(...configs: Partial<ConfigJson>[]): Config
   }, {} as ConfigJson)
 }
 
-export const createContext = async (configPath: string, overrides: ConfigJson): Promise<Context> => {
+export const createContext = async (configPath: string, overrides: ConfigJson, transactionContext: boolean = false): Promise<Context> => {
   const config: ConfigJson = consolidateConfigs(DEFAULT_CONFIG, readConfig(configPath), overrides)
 
-  const signer = await createSignerFromPath(config.keypair!)
+  // if transactionContext is true, we need to make sure we have a valid keypair either in the config or overrides
+  if (transactionContext === true && !config.keypair && !overrides.keypair) {
+    throw new Error('No keypair specified in config or args. Please set a keypair path in your config file or use --keypair flag.')
```

Comment:
as suggested in the comments above, this error should be moved into the createSignerFromPath